### PR TITLE
added keyboard input

### PR DESCRIPTION
**Pull request by Michasssada**
i added a keyboard input to your os. took it from my own os Azbest OS

### CODE REVIEW

**Comment by Szymekk44** [comment]
thats craaxy... but there is no GDT (and IDT) so no interrupts...

**Comment by Michasssada** [comment]
> thats crazy... but there is no GDT (and IDT) so no interrupts...

I know I took it from my OS and I need to rewire it. When I do it I update it here

File: kernel/main.c
Comment (jamjellyfish) [inline kernel/main.c:1]: Probably a better way to run keyboard handling every second, but still better than no PS/2

Comment (jamjellyfish) [reply kernel/main.c:1]: > Probably a better way to run keyboard handling every second, but still better than no PS/2

If this gets merged, next step would to build some sort of IO manager in general. Mouse support is always later in the process and it needs to be tracking both.

Here's some *c psuedocode* for it:
```c
void KERNEL_TRACKING(...) {
        KeyboardHandler();
        MouseHandler();
        // any other things we need to update go here
}
```

**Comment by gorciu-official** [comment]
can you fix confilicts pls?

**Review by gorciu-official — COMMENTED** [review]
i don't have time to decide should i merge but it looks pretty fine

**Comment by jamjellyfish** [comment]
> i don't have time to decide should i merge but it looks pretty fine

Good to merge, getting PS/2 working is a decent step towards terminal interaction

**Comment by jamjellyfish** [comment]
My pull #13 adds this same driver but is slightly modified to fit terminal raw mode requirements. When that is merged, this can be closed.

**Comment by olssonn** [comment]
I fixed the merge conflict because I'm built different

**Pull request merged**